### DISCUSSION TITLE
MaterialGrid for LorentzianSusceptibility materials

### DISCUSSION THREAD

**Discussion by yungchiang** [NanoComp/meep#2037] [Q&A] [answered]
Hello Meep community,

According to the Meep document, LorentzianSusceptibility type materials are supported in MaterialGrid class. However, after doing many times of try and error, I still found that only the epsilon parts are interpolated and the E_susceptibilities are completely ignored. And the geometry region using the MaterialGrid material showed that it is just a pure dielectric material not a LorentzianSusceptibility material. (I have read the code “meepgeom.cpp” and found that sigma_diag and sigma_offdiag are interpolated in the code.) I can find some examples that use MaterialGrid for two dielectric materials in Github. But I can’t find any example for MaterialGrid with two LorentzianSusceptibility materials. Can anyone provide a simple example for MaterialGrid using LorentzianSusceptibility materials? Thank you for your help.

**Comment by stevengj** [comment, marked as answer]
You probably need to add the susceptibilities to `extra_materials` (see the [`Simulation` constructor](https://meep.readthedocs.io/en/latest/Python_User_Interface/#simulation)).

**Reply by yungchiang** [reply]
Thank you for your help. After adding extra_materials, MaterialGrid for LorentzianSusceptibility materials is normal now.

**Comment by kannesito** [comment]
Hi there,
a follow-up question: What happens, if one of the two materials/mp.medium (or maybe even both media) in MaterialGrid has (have) a **D_conductivity** specified? Will the medium with the conductivity retain its conductivity value? 'Cause that would mean, a 3D material of arbitrary shape and Permittivity + D_conductivity could be initialized in the cell, while the second medium could be vacuum or mp.air, do I get this right?
Thanks in advance!
